tests/Microsoft.Windows.Developer/Microsoft.Windows.Developer.Tests.ps1

```diff
@@ -850,6 +856,213 @@ InModuleScope Microsoft.Windows.Developer {
+         @{ FireWallRuleState = [FireWallRuleStateState]::Nonexistent; EnsureState = [Ensure]::Absent }
+         @{ FireWallRuleState = [FireWallRuleStateState]::InDesiredState; EnsureState = [Ensure]::Present }
+         @{ FireWallRuleState = [FireWallRuleStateState]::InDesiredState; EnsureState = [Ensure]::Absent }
+         @{ FireWallRuleState = [FireWallRuleStateState]::NotinDesiredState; EnsureState = [Ensure]::Present }
+         @{ FireWallRuleState = [FireWallRuleStateState]::NotinDesiredState; EnsureState = [Ensure]::Absent }
+      ) {
+         $isInDesiredState = $false
+         if (($FireWallRuleState -eq [FireWallRuleStateState]::InDesiredState) -and ($EnsureState -eq [Ensure]::Present)) {
+            $isInDesiredState = $true
+         } elseif (($FireWallRuleState -eq [FireWallRuleStateState]::Nonexistent) -and ($EnsureState -eq [Ensure]::Absent)) {
+            $isInDesiredState = $true
+         }
+
+         Mock Get-NetFirewallRule {
+            if ($FireWallRuleState -eq [FireWallRuleStateState]::InDesiredState) {
+               return @{
+                  Name        = 'TestRule'
+                  DisplayName = 'Test Rule'
+                  Action      = 'Allow'
+                  Description = 'Test Description'
+                  Direction   = 'Inbound'
+                  Enabled     = $true
+                  Profile     = 'Domain, Private'
+               }
+            } elseif ($FireWallRuleState -eq [FireWallRuleStateState]::NotinDesiredState) {
+               return @{
+                  Name        = 'TestRule'
+                  DisplayName = 'Different Test Rule'
+                  Action      = 'Block'
+                  Description = 'Different Test Description'
+                  Direction   = 'Outbound'
+                  Enabled     = $false
+                  Profile     = ''
+               }
+            } else {
+               return $null
+            }
+         }
+
+         $firewallRule = [FirewallRule]@{
+            Name        = 'TestRule'
+            Ensure      = $EnsureState
+            DisplayName = 'Test Rule'
+            Action      = [Action]::Allow
+            Description = 'Test Description'
+            Direction   = [Direction]::Inbound
+            Enabled     = $true
+            Profiles    = @('Domain', 'Private')
+            LocalPort   = 80, 443
+            Protocol    = 'TCP'
+         }
+
+         $firewallRule.Set()
+
+         $newRuleCount = 0
+         $setRuleCount = 0
+         $removeRuleCount = 0
+
+         if ($isInDesiredState -eq $false) {
+            if ($EnsureState -eq [Ensure]::Present) {
+               if ($FireWallRuleState -eq [FireWallRuleStateState]::Nonexistent) {
+                  $newRuleCount = 1
+               } else {
+                  $setRuleCount = 1
+               }
+            } else {
+               $removeRuleCount = 1
+            }
+         }
+
+         #Excluding LocalPort due to Pester limitations
```

Comment:
I ran into an issue with Pester not comparing arrays correctly, excluding the field for pragmatism.